File: CareConnect/CareConnect.Models/SearchObjects/ClientSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class ClientSearchObject : BaseSearchObject<ClientAdditionalData>
    {
        public string? FirstNameGTE { get; set; }

        public string? LastNameGTE { get; set; }

        public string? Email { get; set; }

        public bool? EmploymentStatus { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/search_objects/client_search_object.dart
import 'package:careconnect_mobile/models/search_objects/client_additional_data.dart';
import 'package:json_annotation/json_annotation.dart';

part 'client_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class ClientSearchObject {
  String? fts;
  String? firstNameGTE;
  String? lastNameGTE;
  String? email;
  bool? employmentStatus;
  int? page;
  String? sortBy;
  bool? sortAscending;
  ClientAdditionalData? additionalData;
  bool? includeTotalCount;

  ClientSearchObject({
    this.fts,
    this.firstNameGTE,
    this.lastNameGTE,
    this.email,
    this.employmentStatus,
    this.page,
    this.sortBy,
    this.sortAscending,
    this.additionalData,
    this.includeTotalCount,
  });

  factory ClientSearchObject.fromJson(Map<String, dynamic> json) =>
      _$ClientSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$ClientSearchObjectToJson(this);
}
